Booktrix/Controllers/SettingsView/CategoriesTableViewController.swift

```diff
@@ -0,0 +1,115 @@
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        fetchCategories()
+    }
+    
+    func fetchCategories() {
+        showHud()
+        viewModel.categoriesIndex(completion: { [weak self] result in
+            switch result {
+            case .success:
+                self?.tableView.reloadData()
+                self?.hideHud()
+            case .failure(let error):
+                self?.showError(title: nil, subtitle: error.errorMessage, dismissDelay: 3.0)
+            }
+        })
+    }
+    
+    // MARK: - Table view data source
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return viewModel.categoriesList.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let category = viewModel.categoriesList[indexPath.row]
+        
+        let cell: CategoryCell = tableView.dequeue()
+        cell.setup(category: category)
+        
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return .leastNonzeroMagnitude
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return .leastNonzeroMagnitude
+    }
+    
+    override func tableView(_ tableView: UITableView, editActionsForRowAt: IndexPath) -> [UITableViewRowAction]? {
+        let edit = UITableViewRowAction(style: .normal, title: LocalizedString.edit) { action, index in
+            self.editCategory(category: self.viewModel.categoriesList[editActionsForRowAt.row])
+        }
+        edit.backgroundColor = UIColor(red:0.30, green:0.53, blue:0.66, alpha:1.0)
+        
+        let delete = UITableViewRowAction(style: .normal, title: LocalizedString.delete) { action, index in
+            self.deleteCategory(category: self.viewModel.categoriesList[editActionsForRowAt.row])
+        }
+        delete.backgroundColor = UIColor(red:0.69, green:0.25, blue:0.24, alpha:1.0)
+        
+        return [edit, delete]
+    }
+    
+    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        return true
+    }
+    
+    func addCategory() {
+        pushViewFromStoryboard(controller: Wireframe.CategoryView().category())
+    }
+    
+    func editCategory(category: Category) {
+        let vc = Wireframe.CategoryView().category() as CategoryViewController
+        vc.viewModel = CategoryViewModel(category: category)
+        pushViewFromStoryboard(controller: vc)
+    }
+    
+    func deleteCategory(category: Category) {
+        showHud()
+        viewModel.deleteCategory(category: category, completion: { [weak self] result in
```

Comment:
lepiej używać skróconej składni, gdy completion podajesz jako ostatni parametr możesz pominąc nazwe argumentu np. `viewModel.deleteCategory(category: category) { [weak self] result in  }`

Z drobniejszych uwag - jak nazywasz metodę `deleteCategory` to lepiej pierwszy argument zrobić nienazwanym bo to wygląda dziwnie, jakbyś się powtarzała. Wtedy zamiast deklarować metodę `deleteCategory(category: Category)` robisz `deleteCategory(_ category: Category)`.